2-mandatory/3-groceries.js

```diff
@@ -27,13 +27,21 @@ Exercise 1:
 */
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
-
+for (let key in weeklyMealPlan) {
```

Comment:
Really good. Try to use meaningful variable names. `key` could be `day` and `items` could be `ingredient`